root/var/lib/machines/nsdc/usr/local/sbin/checkpassword.pl

```diff
@@ -0,0 +1,77 @@
+#!/usr/bin/perl -w
+# This Script will check password complexity
+# Got this script from https://www.mylinuxplace.com/tag/check-password-script/ and adapted it to work...
+$min_length=8;
+$min_upercase=1;
+$min_lowercase=1;
+$min_digits=1;
+$min_specialchar=1;
```

Comment:
We could reflect the system password policy. IIRC it is saved as

```
[root@nethservice ~]# config show passwordstrength 
passwordstrength=configuration
    Admin=strong
    Ibays=strong
    MaxPassAge=180
    MinPassAge=0
    PassExpires=no
    PassWarning=7
    Users=strong
```

Reply:
Good idea.

Reply:
There's a pitfall: you can't use Perl esmith::DB module directly in that script (it runs in the container...)

To implement it, just for instance, you could create/remove a file-flag in this script:

https://github.com/NethServer/nethserver-dc/blob/8882eb2622f962753da8a6a0ebb51e986267dca2/root/etc/e-smith/events/actions/nethserver-dc-password-policy#L84-L90

...then test for existence in checkpassword.pl. If the file-flag does not exist, apply "strong" policy (default).

Reply:
Thanks for the file flag hint.